drake/systems/plants/collision/ResultCollector.h

```diff
@@ -0,0 +1,31 @@
+#ifndef DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+#define DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+
+#include "drake/systems/plants/collision/DrakeCollision.h"
+
+namespace DrakeCollision {
+class ResultCollector {
+ public:
+  virtual ~ResultCollector() {}
+
+  virtual void addPointPairResult(const PointPair& result);
+
+  inline void addSingleResult(const ElementId idA, const ElementId idB,
+                              const Eigen::Vector3d& ptA,
+                              const Eigen::Vector3d& ptB,
+                              const Eigen::Vector3d& normal,
+                              const double distance) {
+    addPointPairResult(PointPair(idA, idB, ptA, ptB, normal, distance));
+  }
```

Comment:
By way of example, the doxygen for `drake::util::CompareMatrices` suggests that the 'const double' _will_ show up in the generated html.  Sad.

While I agree that the const isn't part of the API, so from a principled point of view we should remove it, I'll argue for a more practical answer.

I feel like Drake's disease of putting most of its code in header files means that we should treat header files like implementation -- and so err on the side of making that job easier, and just live with our doxygen API docs having a few warts such as the extra 'const'. It's only odd to people who understand the subtlety we are getting at here, and I hope they will cut us some slack (and not be confused).